hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadata.java

```diff
@@ -224,4 +232,9 @@ HoodieData<HoodieRecord<HoodieMetadataPayload>> getRecordsByKeyPrefixes(List<Str
    * Clear the states of the table metadata.
    */
   void reset();
+
+  /**
+   * Returns the number of shards in a metadata table partition.
+   */
+  int getNumShards(MetadataPartitionType partition);
```

Comment:
shards is a new terminology for hudi in general. For an engineer who comes from data base or data infra background might knew what a shard is. But can we use "getNumFileGroups" so that its in line with our configs like min file groups, max file groups etc.

Reply:
Renamed